src/App.tsx

```diff
@@ -1,61 +1,148 @@
-import './App.scss';
+import "./App.scss";
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useEffect, useState } from "react";
+import usersFromServer from "./api/users";
+import todosFromServer from "./api/todos";
+import { TodoList } from "./components/TodoList";
+import { User } from "./types/user";
+import { Todo } from "./types/todo";
+import { findUserByTodoUserId } from "./helpers/findUserByTodoUserId";
+
+interface FormValue {
+  userId: number;
+  title: string;
+  emptyTitleError: boolean;
+  emptyUserError: boolean;
+}
+
+const defaultFormValue: FormValue = {
+  userId: 0,
+  title: "",
+  emptyTitleError: false,
+  emptyUserError: false,
+};
 
 export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [formValue, setFormValue] = useState<FormValue>(defaultFormValue);
+
+  useEffect(() => {
+    setUsers(usersFromServer);
+    setTodos(todosFromServer);
+  }, []);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const { title, userId } = formValue;
+    let error = false;
+
+    if (!userId) {
+      setFormValue((prevState) => ({ ...prevState, emptyUserError: true }));
+      error = true;
+    }
+
+    if (!title) {
+      setFormValue((prevState) => ({ ...prevState, emptyTitleError: true }));
+      error = true;
+    }
+
+    if (error) {
+      return;
+    }
+
+    const user = findUserByTodoUserId(users, userId);
+
+    if (user) {
+      const newTodo = {
+        id:
+          todos.sort((todoA, todoB) => todoA.id - todoB.id)[todos.length - 1]
+            .id + 1,
+        title: formValue.title,
+        completed: false,
+        userId: userId,
+      };
+
+      setTodos([...todos, newTodo]);
+    }
+
+    setFormValue(defaultFormValue);
+  };
+
+  const { title, userId } = formValue;
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form action="/api/users" method="GET" onSubmit={handleSubmit}>
         <div className="field">
```

Comment:
Remove 'action' and 'method' attributes, since they are not needed when handling the form submission with React